packages/website/plugins/generated-rule-docs.ts

```diff
@@ -236,22 +227,27 @@ export const generatedRuleDocs: Plugin = () => {
     "@typescript-eslint/${file.stem}": "${optionLevel}"
   }
 }`;
-      root.children.splice(firstH2Index, 0, {
-        lang: 'js',
-        type: 'code',
-        meta: 'title=".eslintrc.cjs"',
-        value: `module.exports = ${getEslintrcString};`,
-      } as mdast.Code);
+      children.splice(
+        firstH2Index,
+        0,
+        {
+          lang: 'js',
+          type: 'code',
+          meta: 'title=".eslintrc.cjs"',
+          value: `module.exports = ${getEslintrcString};`,
+        } as mdast.Code,
+        {
+          value: `<try-in-playground eslintrcHash="${convertToPlaygroundHash(
+            getEslintrcString,
+          )}" />`,
+          type: 'jsx',
+        } as unist.Node,
+      );
 
-      root.children.splice(firstH2Index + 1, 0, {
-        value: `<try-in-playground eslintrcHash="${convertToPlaygroundHash(
-          getEslintrcString,
-        )}" />`,
-        type: 'jsx',
-      } as unist.Node);
+      optionsH2Index += 2;
```

Comment:
This is the real fix. Everything else is just nice-to-have and refactoring spaghetti.